README.md
codejam2012
===========

McGill Codejam 2012

Team RuntimeException

By Nguyen Khoi Tran and Alan Yang (But really, mostly Alan)

![ScreenShot](https://raw.github.com/humar104/codejam2012/master/screen_shot_program.jpg)


GUI/History.txt
Time            Price           Type
23              9.215            sell
25              9.215            sell
27              9.185            sell
32              9.185            sell
55              9.151            buy
55              9.151            buy
56              9.186            buy
60              9.246            buy
71              9.218            sell
73              9.188            sell
74              9.188            sell
74              9.218            buy
79              9.219            buy
79              9.204            sell
82              9.219            buy
82              9.249            buy
100              9.251            sell
102              9.251            sell
104              9.251            sell
108              9.317            buy
107              9.317            buy
108              9.317            buy
121              9.283            sell
125              9.319            sell
126              9.349            buy
127              9.349            buy
159              9.382            sell
162              9.413            sell
164              9.413            sell
166              9.448            buy
168              9.383            sell
171              9.414            sell
175              9.449            buy
174              9.449            buy
176              9.449            buy
179              9.450            buy
181              9.385            sell
182              9.385            sell
182              9.415            sell
184              9.385            sell
186              9.450            buy
185              9.450            buy
186              9.450            buy
189              9.451            buy
192              9.416            sell
193              9.416            sell
193              9.416            sell
197              9.416            sell
211              9.388            buy
213              9.328            sell
216              9.388            buy
216              9.359            buy
217              9.359            sell
217              9.359            sell
218              9.359            buy
220              9.359            sell
236              9.361            buy
236              9.391            buy
236              9.391            buy
240              9.391            buy
248              9.392            sell
251              9.392            sell
252              9.422            buy
253              9.392            sell
256              9.423            buy
256              9.423            buy
257              9.363            sell
257              9.363            sell
258              9.363            sell
262              9.393            sell
262              9.423            buy
264              9.424            buy
264              9.424            buy
267              9.424            buy
271              9.394            sell
275              9.425            buy
294              9.552            sell
296              9.552            sell
300              9.523            sell
301              9.553            buy
303              9.588            buy
304              9.588            buy
330              9.590            sell
331              9.590            sell
332              9.590            sell
332              9.620            buy
334              9.680            buy
335              9.680            buy
341              9.559            sell
342              9.559            sell
342              9.683            buy
343              9.558            sell
344              9.558            sell
346              9.522            sell
362              9.555            buy
362              9.490            buy
363              9.585            buy
367              9.585            buy
376              9.585            sell
378              9.615            buy
391              9.586            sell
392              9.586            sell
393              9.586            sell
396              9.521            sell
401              9.616            buy
402              9.556            sell
407              9.586            buy
407              9.556            sell
422              9.556            buy
422              9.586            buy
422              9.586            buy
427              9.556            buy
430              9.556            sell
431              9.556            sell
432              9.521            sell
434              9.521            sell
437              9.586            buy
436              9.586            buy
438              9.521            sell
439              9.521            sell
440              9.586            buy
441              9.586            buy
440              9.586            buy
441              9.586            buy
441              9.586            sell
442              9.616            buy
453              9.557            sell
454              9.557            sell
454              9.587            sell
454              9.587            buy
459              9.557            sell
461              9.587            buy
463              9.587            buy
463              9.587            sell
464              9.587            buy
465              9.587            buy
477              9.647            sell
479              9.617            sell
481              9.682            buy
482              9.682            buy
482              9.647            sell
490              9.647            sell
491              9.647            sell
491              9.682            buy
492              9.647            sell
494              9.647            sell
497              9.647            sell
497              9.682            buy
499              9.647            sell
500              9.682            buy
499              9.682            buy

GUI/ui_rte.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'RTEDlg.ui'
#
# Created: Sun Nov 18 22:46:19 2012
#      by: PyQt4 UI code generator 4.9.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    _fromUtf8 = lambda s: s

class Ui_RTEDlg(object):
    def setupUi(self, RTEDlg):
        RTEDlg.setObjectName(_fromUtf8("RTEDlg"))
        RTEDlg.resize(1222, 572)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(RTEDlg.sizePolicy().hasHeightForWidth())
        RTEDlg.setSizePolicy(sizePolicy)
        RTEDlg.setMinimumSize(QtCore.QSize(1222, 572))
        RTEDlg.setMaximumSize(QtCore.QSize(1222, 572))
        RTEDlg.setWhatsThis(_fromUtf8(""))
        self.line = QtGui.QFrame(RTEDlg)
        self.line.setGeometry(QtCore.QRect(260, 10, 20, 551))
        self.line.setFrameShape(QtGui.QFrame.VLine)
        self.line.setFrameShadow(QtGui.QFrame.Sunken)
        self.line.setObjectName(_fromUtf8("line"))
        self.smaMpl = MatplotlibWidget(RTEDlg)
        self.smaMpl.setGeometry(QtCore.QRect(280, 20, 461, 271))
        self.smaMpl.setObjectName(_fromUtf8("smaMpl"))
        self.emaMpl = MatplotlibWidget(RTEDlg)
        self.emaMpl.setGeometry(QtCore.QRect(280, 290, 461, 271))
        self.emaMpl.setObjectName(_fromUtf8("emaMpl"))
        self.lwmaMpl = MatplotlibWidget(RTEDlg)
        self.lwmaMpl.setGeometry(QtCore.QRect(750, 20, 461, 271))
        self.lwmaMpl.setObjectName(_fromUtf8("lwmaMpl"))
        self.tmaMpl = MatplotlibWidget(RTEDlg)
        self.tmaMpl.setGeometry(QtCore.QRect(750, 290, 461, 271))
        self.tmaMpl.setObjectName(_fromUtf8("tmaMpl"))
        self.groupBox = QtGui.QGroupBox(RTEDlg)
        self.groupBox.setGeometry(QtCore.QRect(10, 10, 251, 141))
        self.groupBox.setObjectName(_fromUtf8("groupBox"))
        self.gridLayout_2 = QtGui.QGridLayout(self.groupBox)
        self.gridLayout_2.setObjectName(_fromUtf8("gridLayout_2"))
        self.gridLayout_3 = QtGui.QGridLayout()
        self.gridLayout_3.setObjectName(_fromUtf8("gridLayout_3"))
        self.connectButton = QtGui.QPushButton(self.groupBox)
        self.connectButton.setObjectName(_fromUtf8("connectButton"))
        self.gridLayout_3.addWidget(self.connectButton, 0, 0, 1, 1)
        self.disconnectButton = QtGui.QPushButton(self.groupBox)
        self.disconnectButton.setObjectName(_fromUtf8("disconnectButton"))
        self.gridLayout_3.addWidget(self.disconnectButton, 0, 1, 1, 1)
        self.gridLayout_2.addLayout(self.gridLayout_3, 0, 0, 1, 1)
        self.gridLayout = QtGui.QGridLayout()
        self.gridLayout.setObjectName(_fromUtf8("gridLayout"))
        self.label_3 = QtGui.QLabel(self.groupBox)
        self.label_3.setObjectName(_fromUtf8("label_3"))
        self.gridLayout.addWidget(self.label_3, 0, 0, 1, 1)
        self.ipEdit = QtGui.QLineEdit(self.groupBox)
        self.ipEdit.setObjectName(_fromUtf8("ipEdit"))
        self.gridLayout.addWidget(self.ipEdit, 0, 1, 1, 1)
        self.pfLabel = QtGui.QLabel(self.groupBox)
        self.pfLabel.setWordWrap(True)
        self.pfLabel.setObjectName(_fromUtf8("pfLabel"))
        self.gridLayout.addWidget(self.pfLabel, 1, 0, 1, 1)
        self.pfEdit = QtGui.QLineEdit(self.groupBox)
        self.pfEdit.setObjectName(_fromUtf8("pfEdit"))
        self.gridLayout.addWidget(self.pfEdit, 1, 1, 1, 1)
        self.tbLabel = QtGui.QLabel(self.groupBox)
        self.tbLabel.setWordWrap(True)
        self.tbLabel.setObjectName(_fromUtf8("tbLabel"))
        self.gridLayout.addWidget(self.tbLabel, 2, 0, 1, 1)
        self.tbEdit = QtGui.QLineEdit(self.groupBox)
        self.tbEdit.setObjectName(_fromUtf8("tbEdit"))
        self.gridLayout.addWidget(self.tbEdit, 2, 1, 1, 1)
        self.gridLayout_2.addLayout(self.gridLayout, 1, 0, 1, 1)
        self.groupBox_2 = QtGui.QGroupBox(RTEDlg)
        self.groupBox_2.setGeometry(QtCore.QRect(10, 160, 251, 61))
        self.groupBox_2.setObjectName(_fromUtf8("groupBox_2"))
        self.verticalLayout_3 = QtGui.QVBoxLayout(self.groupBox_2)
        self.verticalLayout_3.setObjectName(_fromUtf8("verticalLayout_3"))
        self.verticalLayout_2 = QtGui.QVBoxLayout()
        self.verticalLayout_2.setObjectName(_fromUtf8("verticalLayout_2"))
        self.startButton = QtGui.QPushButton(self.groupBox_2)
        self.startButton.setObjectName(_fromUtf8("startButton"))
        self.verticalLayout_2.addWidget(self.startButton)
        self.verticalLayout_3.addLayout(self.verticalLayout_2)
        self.sendButton = QtGui.QPushButton(RTEDlg)
        self.sendButton.setGeometry(QtCore.QRect(20, 240, 231, 61))
        self.sendButton.setObjectName(_fromUtf8("sendButton"))
        self.serialLabel = QtGui.QLabel(RTEDlg)
        self.serialLabel.setGeometry(QtCore.QRect(290, 0, 911, 31))
        self.serialLabel.setWordWrap(True)
        self.serialLabel.setObjectName(_fromUtf8("serialLabel"))
        self.label = QtGui.QLabel(RTEDlg)
        self.label.setGeometry(QtCore.QRect(430, 30, 261, 16))
        self.label.setObjectName(_fromUtf8("label"))
        self.label_2 = QtGui.QLabel(RTEDlg)
        self.label_2.setGeometry(QtCore.QRect(420, 300, 221, 16))
        self.label_2.setObjectName(_fromUtf8("label_2"))
        self.label_4 = QtGui.QLabel(RTEDlg)
        self.label_4.setGeometry(QtCore.QRect(890, 30, 231, 16))
        self.label_4.setObjectName(_fromUtf8("label_4"))
        self.label_5 = QtGui.QLabel(RTEDlg)
        self.label_5.setGeometry(QtCore.QRect(880, 300, 231, 16))
        self.label_5.setObjectName(_fromUtf8("label_5"))
        self.label_6 = QtGui.QLabel(RTEDlg)
        self.label_6.setGeometry(QtCore.QRect(10, 340, 221, 111))
        self.label_6.setTextFormat(QtCore.Qt.AutoText)
        self.label_6.setObjectName(_fromUtf8("label_6"))

        self.retranslateUi(RTEDlg)
        QtCore.QMetaObject.connectSlotsByName(RTEDlg)

    def retranslateUi(self, RTEDlg):
        RTEDlg.setWindowTitle(QtGui.QApplication.translate("RTEDlg", "Algorithmic Trading Application", None, QtGui.QApplication.UnicodeUTF8))
        self.groupBox.setTitle(QtGui.QApplication.translate("RTEDlg", "Connection Details", None, QtGui.QApplication.UnicodeUTF8))
        self.connectButton.setText(QtGui.QApplication.translate("RTEDlg", "&Connect", None, QtGui.QApplication.UnicodeUTF8))
        self.disconnectButton.setText(QtGui.QApplication.translate("RTEDlg", "&Disconnect", None, QtGui.QApplication.UnicodeUTF8))
        self.label_3.setText(QtGui.QApplication.translate("RTEDlg", "IP Address", None, QtGui.QApplication.UnicodeUTF8))
        self.pfLabel.setText(QtGui.QApplication.translate("RTEDlg", "Price Feed Port", None, QtGui.QApplication.UnicodeUTF8))
        self.pfEdit.setText(QtGui.QApplication.translate("RTEDlg", "8211", None, QtGui.QApplication.UnicodeUTF8))
        self.tbLabel.setText(QtGui.QApplication.translate("RTEDlg", "Trade Booking Port", None, QtGui.QApplication.UnicodeUTF8))
        self.tbEdit.setText(QtGui.QApplication.translate("RTEDlg", "8212", None, QtGui.QApplication.UnicodeUTF8))
        self.groupBox_2.setTitle(QtGui.QApplication.translate("RTEDlg", "GUI Commands", None, QtGui.QApplication.UnicodeUTF8))
        self.startButton.setText(QtGui.QApplication.translate("RTEDlg", "Start", None, QtGui.QApplication.UnicodeUTF8))
        self.sendButton.setText(QtGui.QApplication.translate("RTEDlg", "Send Report", None, QtGui.QApplication.UnicodeUTF8))
        self.serialLabel.setText(QtGui.QApplication.translate("RTEDlg", "Ceremony ID", None, QtGui.QApplication.UnicodeUTF8))
        self.label.setText(QtGui.QApplication.translate("RTEDlg", "Simple Moving Average(SMA)", None, QtGui.QApplication.UnicodeUTF8))
        self.label_2.setText(QtGui.QApplication.translate("RTEDlg", "Exponential Moving Average(EMA)", None, QtGui.QApplication.UnicodeUTF8))
        self.label_4.setText(QtGui.QApplication.translate("RTEDlg", "Linear Weighted Moving Average(LWMA)", None, QtGui.QApplication.UnicodeUTF8))
        self.label_5.setText(QtGui.QApplication.translate("RTEDlg", "Triangular Moving Average(TMA)", None, QtGui.QApplication.UnicodeUTF8))
        self.label_6.setText(QtGui.QApplication.translate("RTEDlg", "<html><head/><body><p>Legend[MA = Moving Average]</p><p>Price: Blue Line</p><p>Slow MA: Red Line (Period[N] = 20)</p><p>Fast MA: Green Line (Period[N] = 5)</p></body></html>", None, QtGui.QApplication.UnicodeUTF8))

from matplotlibwidget import MatplotlibWidget
